client/app/telecom/pack/xdsl/access/pack-xdsl-access.controller.js

```diff
@@ -1,251 +1,277 @@
-angular.module('managerApp').controller('XdslAccessCtrl', function (
-  $scope, $stateParams, $uibModal, $q, $filter, $translate, $templateCache,
-  OvhApiXdsl, OvhApiXdslTasksCurrent, OvhApiXdslLines, OvhApiXdslNotifications, OvhApiXdslModem,
-  OvhApiXdslIps, OvhApiPackXdsl, Toast, ToastError, PACK_IP, REDIRECT_URLS,
-) {
-  const self = this;
-
-  this.packName = $stateParams.packName;
-
-  $scope.loaders = {
-    details: true,
-    tasks: true,
-    deconsolidation: true,
-    xdsl: true,
-  };
-
-  $scope.access = {
-    xdsl: null,
-    tasks: { current: {} },
-  };
-
-  $scope.constants = {
-    rangeOfBaseIpv4IP: PACK_IP.baseIpv4Range,
-  };
-
-  function init() {
-    self.number = $stateParams.number;
-    self.getLinesDetails();
+angular.module('managerApp').controller('XdslAccessCtrl', class XdslAccessCtrl {
+  constructor(
+    $filter, $q, $scope, $stateParams, $templateCache, $translate, $uibModal,
+    OvhApiPackXdsl, OvhApiXdsl, OvhApiXdslIps, OvhApiXdslLines, OvhApiXdslModem,
+    OvhApiXdslNotifications, OvhApiXdslTasksCurrent, Toast, ToastError,
+    PACK, PACK_IP, REDIRECT_URLS,
+  ) {
+    this.$filter = $filter;
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$stateParams = $stateParams;
+    this.$templateCache = $templateCache;
+    this.$translate = $translate;
+    this.$uibModal = $uibModal;
+    this.OvhApiPackXdsl = OvhApiPackXdsl;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.OvhApiXdslIps = OvhApiXdslIps;
+    this.OvhApiXdslLines = OvhApiXdslLines;
+    this.OvhApiXdslModem = OvhApiXdslModem;
+    this.OvhApiXdslNotifications = OvhApiXdslNotifications;
+    this.OvhApiXdslTasksCurrent = OvhApiXdslTasksCurrent;
+    this.Toast = Toast;
+    this.ToastError = ToastError;
+    this.PACK = PACK;
+    this.PACK_IP = PACK_IP;
+    this.REDIRECT_URLS = REDIRECT_URLS;
+  }
+
+  $onInit() {
+    this.packName = this.$stateParams.packName;
+
+    this.$scope.loaders = {
```

Comment:
Can't you remove the `$scope` entirely ?

Reply:
There are a few `ngInclude` hanging around in the associated view so this could take a bit of time to refactor properly

Reply:
I was suspecting that